### ISSUE TITLE
Black formatting - output: nil

### ISSUE THREAD

**Issue by kronomax** [jose-elias-alvarez/null-ls.nvim#618]
### FAQ

- [X] I have checked the [FAQ](https://github.com/jose-elias-alvarez/null-ls.nvim#faq) and it didn't resolve my problem.

### Issues

- [X] I have checked [existing issues](https://github.com/jose-elias-alvarez/null-ls.nvim/issues?q=is%3Aissue) and there are no issues with the same problem.

### Neovim Version

0.6.0

### Operating System

Windows 10 with nvim running on WSL2 - Ubuntu 20.04

### Minimal config

-- this template is borrowed from nvim-lspconfig
local on_windows = vim.loop.os_uname().version:match("Windows")

local function join_paths(...)
    local path_sep = on_windows and "\\" or "/"
    local result = table.concat({ ... }, path_sep)
    return result
end

vim.cmd([[set runtimepath=$VIMRUNTIME]])

local  temp_dir
if on_windows then
    temp_dir = vim.loop.os_getenv("TEMP")
else
    temp_dir = "/tmp"
end

vim.cmd("set packpath=" .. join_paths(temp_dir, "nvim", "site"))

local package_root = join_paths(temp_dir, "nvim", "site", "pack")
local install_path = join_paths(package_root, "packer", "start", "packer.nvim")
local compile_path = join_paths(install_path, "plugin", "packer_compiled.lua")

local null_ls_config = function()
    local null_ls = require("null-ls")
    -- add only what you need to reproduce your issue
    null_ls.setup({
        sources = {null_ls.builtins.formatting.black},
        debug = true,
    })
end

local function load_plugins()
    -- only add other plugins if they are necessary to reproduce the issue
    require("packer").startup({
        {
            "wbthomason/packer.nvim",
            {
                "jose-elias-alvarez/null-ls.nvim",
                requires = { "nvim-lua/plenary.nvim" },
                config = null_ls_config,
            },
        },
        config = {
            package_root = package_root,
            compile_path = compile_path,
        },
    })
end

if vim.fn.isdirectory(install_path) == 0 then
    vim.fn.system({ "git", "clone", "https://github.com/wbthomason/packer.nvim", install_path })
    load_plugins()
    require("packer").sync()
else
    load_plugins()
    require("packer").sync()
end

### Steps to reproduce

1. Open any python file and modify it to have some issues
2. Run the command `:lua vim.lsp.buf.formatting_sync()`

### Expected behavior

The black formatter runs, and formats the file.

### Actual behavior

First, for some reason I was unable to get the minimal_init.lua working. I curled the file and ran `nvim --clean -u minimal_init.lua` and got some errors when nvim opened, it tells me that `module 'packer' not found`. As far as I can read the code, it should have installed packer, but for some reason it's erroring out. If I can figure out how to get the minimal config going, I'd be more than glad to give it another try.

Actual behavior: nothing happens. Debug log has some errors related to it. I can run for example `!black %` and reopen the file, and it has been formatted successfully. I'm only having this issue with black, prettier works just fine with html/js files. Before this I was using `mhartington/formatter.nvim` and it worked just fine with black. `:checkhealth` tells me black is installed properly as well. 

### Debug log

[DEBUG Wed Jan 26 14:18:53 2022] ...t/null-ls.nvim/lua/null-ls/helpers/generator_factory.lua:335: spawning command [black] at /home/raymond with args { "--stdin-filename", "/home/raymond/new.py", "--quiet", "-" }
[TRACE Wed Jan 26 14:18:53 2022] ...t/null-ls.nvim/lua/null-ls/helpers/generator_factory.lua:224: error output: nil
[TRACE Wed Jan 26 14:18:53 2022] ...t/null-ls.nvim/lua/null-ls/helpers/generator_factory.lua:225: output: nil
[TRACE Wed Jan 26 14:19:52 2022] .../site/pack/packer/start/null-ls.nvim/lua/null-ls/rpc.lua:118: received LSP request for

### Help

No

### Requirements

- [X] I have read and followed the instructions above and understand that my issue will be closed if I did not provide the required information.

**Comment by jose-elias-alvarez** [comment]
`minimal_init.lua` may be erroring out because you already have a `/tmp/nvim` directory, which means you're not cloning Packer onto the install path. Could you try deleting the directory (assuming that it's in the same location on WSL) and try again? 

On my end, loading with your config and running `vim.lsp.buf.formatting_sync()` correctly formats a sample Python file. `black --version` outputs `black, 21.12b0 (compiled: no)`, so this may be a version mismatch.

**Comment by kronomax** [comment]
I indeed had an nvim directory in tmp, which prevented the minimal config from working. Removed the conflict, still nothing happened after running `vim.lsp.buf.formatting_sync()`

As per your instructions, trying out black from the command line revealed the issue. I had black installed from apt, and the version was 19-something so it didn't support the `--stdin-filename` flag yet. Updated black, and all is good in the world again. Thanks a lot for the quick help, and for teaching some better debugging skills 😀!
